src/main/java/org/fairdatapoint/config/BootstrapConfig.java

```diff
@@ -74,16 +86,21 @@ public Jackson2RepositoryPopulatorFactoryBean repositoryPopulator() {
             try {
                 // collect fixture resources
                 final Path fixturesPath = dbFixturesPath.resolve("*.json");
-                final Resource[] resources = resourceResolver.getResources("file:" + fixturesPath);
+                resources.addAll(List.of(resourceResolver.getResources("file:" + fixturesPath)));
+                // remove resources that have been applied already
+                // todo: add a "force" config property that forces all fixtures to be (re)applied?
+                //  or just rely on manual edit of the fixture history table?
+                final List<String> appliedFixtures = fixtureHistoryRepository.findAll().stream()
+                        .map(FixtureHistory::getFilename).toList();
+                final List<Resource> resourcesToSkip = resources.stream()
+                        .filter(resource -> appliedFixtures.contains(resource.getFilename())).toList();
+                resources.removeAll(resourcesToSkip);
```

Comment:
why not directly creating a filtered list of resources that are not in `appliedFixtures` directly?

Reply:
That would be more concise, and I started with something like that, but later (in 0f89461) decided to separate it so we can log the number of fixtures applied vs skipped (L101-L102). Could use the `appliedFixtures` count for that as well, but then there could --although unlikely-- be a discrepancy between the actual files (resources) and what is in the repository.